### PR TITLE
staticGetContentTypeByExtension is now a static function

### PR DESCRIPTION
**Pull request by matiasperrone**
staticGetContentTypeByExtension is now a static function, so when called as static no E_DEPRECATED warning is thrown.
Change PEAR::isError to (new PEAR)->isError.

### CODE REVIEW

File: lib/OA/Creative/File.php [hunk 1]
```diff
@@ -155,7 +154,7 @@ function &factory($filePath, $fileName = '')
         $oCreative = new $className($fileName);
 
         $result = $oCreative->loadFile($filePath);
-        if (PEAR::isError($result)) {
+        if ((new PEAR)->isError($result)) {
```
Comment (mbeccati) [inline lib/OA/Creative/File.php:157]: Hmm, there might be lots of `PEAR::isError`, I don't think we want to create an instance of PEAR every time.

File: lib/OA/Creative/File.php [hunk 2]
```diff
@@ -183,7 +182,7 @@ function &factoryString($fileName, $content)
         }
 
         // Get new instance
-        $oCreative = &OA_Creative_File::factory($filePath, $fileName);
+        $oCreative = $this->factory($filePath, $fileName);
```
Comment (mbeccati) [inline lib/OA/Creative/File.php:185]: Factory methods are normally static.

File: lib/OA/Creative/File.php (hunk 1 above)
Comment (matiasperrone) [reply lib/OA/Creative/File.php:157]: This has serious implications in future PHP versions. 
This type of calls are deprecated in 7.0 (https://www.php.net/manual/en/migration70.deprecated.php#migration70.deprecated.static-calls), probably removed in v8.

File: lib/OA/Creative/File.php (hunk 2 above)
Comment (matiasperrone) [reply lib/OA/Creative/File.php:185]: But they are not in the **OA_Creative_File** Class. Do you want that I change the factory method instead?

**Pull request closed without merging**